### PR TITLE
Ability to save `spawnpoint.last_seen`

### PR DESCRIPTION
**Pull request by lenisko**
## Description

**MIGRATION FILE `79.sql` IS RUN !!**

Save spawnpoint encounters timestamp in `spawnpoint.last_seen`

By default, we won't save them on all encounters, only when spawnpoint update method is called.

To enable saving set `SAVE_SPAWNPOINT_LASTSEEN: 1`
When enabled, spawnpoint `last_seen` will be updated only once per 15min to reduce the amount of queries.

## Motivation and Context

Ability to filter out unused spawnpoints.

## How Has This Been Tested?

Running on prod.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by lenisko** [comment]
Open for discussion for now.

**Comment by jfberry** [comment]
Does this cause a database update for every spawnpoint every time a mon is seen at that spawnpoint? That's a lot of updates.  You may be able to reduce the writes by adding to the where clause to only update every so often.

**Comment by lenisko** [comment]
@jfberry Yeah, I was thinking about adding a check, so it will save to `last_seen` only when the difference is greater than an hour. We already pull Spawnpoint data in there, so should be good enough.

File: Sources/RealDeviceMapLib/Modell/SpawnPoint.swift [hunk 1]
```diff
@@ -109,6 +117,37 @@ public class SpawnPoint: JSONConvertibleObject {
 
     }
 
+    public static func setLastSeen(mysql: MySQL?=nil, spawnId: UInt64, oldLastSeen: UInt32) throws {
+
+        let now = UInt32(Date().timeIntervalSince1970)
+
+        if oldLastSeen + 3600 > now {
+            return
```
Comment (lenisko) [inline Sources/RealDeviceMapLib/Modell/SpawnPoint.swift:125]: TODO: Test if we hit this check.

Comment (Fabio1988) [reply Sources/RealDeviceMapLib/Modell/SpawnPoint.swift:125]: why do you distinguish between lastSeen and updated? should be almost the same timestamp, isn't it?
you can update the `save` method to improve the save logic

**Review by Fabio1988 — CHANGES_REQUESTED** [review]

**Review by Fabio1988 — COMMENTED** [review]
RDM does not do anything related to that logic, how is your plan to handle that `last_seen`?`Why updating every 15 minutes?! updating every hour should be enough. I'm aware of unnecessary DB queries

**Review by Fabio1988 — APPROVED** [review]

**Comment by lenisko** [comment]
@Fabio1988 For now collection of data. In the future, we can add ability to maps to filter out outdated spawnpoints (for example, to show only the ones within a day/week/month or so). I'm collecting database metrics on my production to see the impact, but updating data every 15 minutes should flatten queries. As a whole, I saw no impact on my server CPU usage or IO.

As a reminder, this feature will save the data along with newly created or updated spawnpoints every time. But updates on each encounter have to be enabled by `SAVE_SPAWNPOINT_LASTSEEN` flag.

**Comment by lenisko** [comment]
I've collected a metrics from MySQL instance for last few days.
Checked for disabled and enabled saving.I saw no impact on: number of connections, slow queries, table locks, select types, sorts.


Handler ops/s show no visible impact as well (first mark - saving disabled, second - enabled)
2022-03-21_15-17-26-firefox

**Pull request merged**